File: go/tornado.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
	"unicode/utf8"
)

func main() {
	// var text string
	var lineCount int
	reader := bufio.NewReader(os.Stdin)
	fmt.Print("Enter some text: ")
	text, _ := reader.ReadString('\n')
	text = strings.TrimSpace(text)
	if !(utf8.RuneCountInString(text) >= 1) {
		fmt.Print("ERROR: Enter some text!\n")
		main()
	}
	fmt.Print("Enter line amount (-1 for infinite): ")
	_, ierr := fmt.Scanf("%d", &lineCount)
	if ierr != nil {
		fmt.Printf("[%s]: ERROR: Enter a line amount!\n", ierr)
		main()
	}

	if string(text[utf8.RuneCountInString(text)-1]) != " " {
		text = strings.Join([]string{text, " "}, "")
	}

	fmt.Println(text)
	if lineCount != -1 {
		for i := 0; i < lineCount; i++ {
			text = shift(text)
			fmt.Println(text)
		}
	} else {
		for {
			text = shift(text)
			fmt.Println(text)
		}
	}
	os.Exit(0)
}

func shift(str string) string {
	return strings.Join([]string{string(str[1:]), string(str[0])}, "")
}

File: php/tornado.php
#!/usr/bin/env php
<?php
function shift($s) {
    return mb_substr($s, 1) . mb_substr($s, 0, 1);
}
$text = false;
while($text === false) {
    echo 'Enter some text:',"\n";
    $text = trim(fgets(STDIN));
    if($text === "") $text = false;
}
$default = mb_strlen($text)+1;
$count = false;
while($count === false) {
    echo 'Enter line amount (-1 for ∞, default: ',$default,'):',"\n";
    $count = intval(trim(fgets(STDIN)));
    if($count === 0)  $count = $default;
    if($count === -1) $count = PHP_INT_MAX;
    if($count < -1)   $count = false;
}
for($i = 0; $i < $count; $i++) {
    echo $text,"\n";
    $text = shift($text);
    if($i === PHP_INT_MAX) $i = 0;
}
exit(0);
